packages/theme-language-server-common/src/rename/providers/LiquidVariableRenameProvider.spec.ts

```diff
@@ -98,14 +104,66 @@ describe('LiquidVariableRenameProvider', () => {
       assert(result.documentChanges);
       expect((result.documentChanges[0] as TextDocumentEdit).edits).to.applyEdits(
         textDocument,
-        `{% assign animal = 'dog' %}
+        `{% doc %}@param {string} food - favorite food{% enddoc %}
+{% assign animal = 'dog' %}
 {% assign plant = 'cactus' %}
 {% liquid
   echo plant
+  echo food
 %}
 {% assign famous_painting = 'mona lisa' %}
 {% assign paintings = 'starry night, sunday afternoon, the scream' %}
-<p>I have a cool animal and a great plant</p>`,
+<p>I have a cool animal, a great plant, and my favorite food</p>`,
+      );
+    });
+
+    it('returns new name after liquid doc param is renamed in doc', async () => {
+      const params = {
+        textDocument,
+        position: Position.create(0, 26),
+        newName: 'meal',
+      };
+
+      const result = await provider.rename(params);
+      assert(result);
+      assert(result.documentChanges);
+      expect((result.documentChanges[0] as TextDocumentEdit).edits).to.applyEdits(
+        textDocument,
+        `{% doc %}@param {string} meal - favorite food{% enddoc %}
+{% assign animal = 'dog' %}
+{% assign plant = 'cactus' %}
+{% liquid
+  echo plant
+  echo meal
+%}
+{% assign painting = 'mona lisa' %}
+{% assign paintings = 'starry night, sunday afternoon, the scream' %}
+<p>I have a cool animal, a great plant, and my favorite food</p>`,
+      );
+    });
+
+    it('returns new name after liquid doc param is renamed on variable usage', async () => {
+      const params = {
+        textDocument,
+        position: Position.create(5, 9),
+        newName: 'meal',
+      };
+
+      const result = await provider.rename(params);
+      assert(result);
+      assert(result.documentChanges);
+      expect((result.documentChanges[0] as TextDocumentEdit).edits).to.applyEdits(
+        textDocument,
+        `{% doc %}@param {string} meal - favorite food{% enddoc %}
+{% assign animal = 'dog' %}
+{% assign plant = 'cactus' %}
+{% liquid
+  echo plant
+  echo meal
+%}
+{% assign painting = 'mona lisa' %}
+{% assign paintings = 'starry night, sunday afternoon, the scream' %}
+<p>I have a cool animal, a great plant, and my favorite food</p>`,
```

Comment:
Can we add a test to document this behaviour:
- renaming liquidDoc param does not rename variables with same name that are declared within a self-contained scope (if you `{% for food in X %}` in the snippet but have `doc` in the LiquidDOc header)

+ vice versa?